buildlogger/basic_sender.go

```diff
@@ -141,14 +142,18 @@ func (opts *LoggerOptions) validate() error {
 	}
 
 	if opts.ClientConn == nil {
-		if opts.RPCAddress == "" {
-			return errors.New("must specify a RPC address when a client connection is not provided")
+		if opts.BaseAddress == "" || opts.RPCPort == "" {
+			return errors.New("must specify a base address and rpc port when a client connection is not provided")
 		}
-		if !opts.Insecure && (opts.CAFile == "" || opts.CertFile == "" || opts.KeyFile == "") {
-			return errors.New("must specify credential files when making a secure connection over RPC")
+		if !opts.Insecure && (opts.Username == "" || opts.APIKey == "") {
```

Comment:
Are we not going to keep the option open to use certs instead of username/API key? Using username/API key will still have to make a REST request to retrieve the certs using `DialCedar`.

Reply:
I don't think it is actually worth it. Evergreen is the only user and ends up calling DialCedar with its credentials and then passes in the resulting grpc client conn. Resmoke will also end up doing something similar.